content/blog/package-react-components/index.md

```diff
@@ -0,0 +1,776 @@
+for example if two libraries need to serve `images/dog.jpg`.
+
+_Component Authors_ might make the module available via a CDN and that might
+remove the need to fiddle around with CSS and Images but you will need to
+access the library via a Browser Global like the good old days.
+
+### Inline
+
+#### As Component Authors
+
+The bundler setup is about the same as with **Not-inline**.
+You wont be able to use [create-react-library][105] but you can probably
+find other tools that allow you to customize them to Inline everything.
+
+The Webpack configuration is actually not that hard and a working example
+will be provided at the end of this post.
+
+The key Webpack configuration is the inclusion of [url-loader][11] to
+inline images and files right into the final Javascript bundle.
+
+If you use **CSS-in-JS** libraries such as [Emotion][5] or [Styled-components][6] then
+you will be able to use all the CSS features, plus compose them via Javascript and the
+**Theming** experience will be awesome.
+You can easily create a Run-Time default Theme and provide
+hooks for _Component Consumers_ to easily change it at Run-Time.
+
+For larger Component Libraries you will have a greater experience with writing
+and maintaining CSS. You will avoid completely some CSS exclusive problems such as:
+
+- Global Scope and rule collisions
+- Specificity (inline-styles have maximum specificity, **CSS-in-JS** libraries simulate this)
+- Style composition, I believe that regular Javascript object composition is better than CSS cascading global composition model, plus you can create a custom system that fits your team and your business logic, remember, it's regular Javascript. You can easily publish your Themes as private NPM packages internally to shared them among Applications.
+
+While regular inline-styles will go a long way, you will most likely
+need features that inline-styles do not provide and that regular CSS do such
+as pseudo selectors like `:hover` and media queries.
+To solve this problem and provide better Developer Experience overall is that
+**CSS-in-JS** exist.
+
+I recommend going for [Emotion][5] or [Styled-components][6] since they
+are very mature and feature rich, have
+good documentation and an active community.
+
+#### As Component Consumers
+
+Here is where the **Inline** approach _shines_.
+To use a module with all its styles and assets such as images
+you simply need to write a single line of code:
+
+```javascript
+import module from "module"
+```
+
+That is all!
+
+If you want to import the module dynamically (to parallelize and deffer its loading
+in the client) the experience is very similar:
+
+```javascript
+// Inside some function or component
+const module = await import("module")
+```
+
+If you have Webpack configured correctly to create chunks that can be
+dynamically loaded, this is called `Code Splitting`, then `module`
+will be an independent file lazy loaded, only and when its needed.
```

Comment:
_It's_ instead of _its_ (also I think "only and when" sounds a little strange IMHO)

Reply:
NSHO (not so humble opinion).

Fixed!